addons/test_mail/static/tests/activity_tests.js

```diff
@@ -658,6 +659,116 @@ QUnit.module("test_mail", {}, function () {
         }
     );
 
+    QUnit.test(
+        "activity view: 'scheduleActivity' does not add activity_ids condition as selectCreateDialog domain",
+        async function (assert) {
+            patchWithCleanup(ActivityController.prototype, {
+                scheduleActivity() {
+                    this._super();
+                    assert.step(JSON.stringify(this.props.domain));
+                },
+            });
+            Object.assign(serverData.views, {
+                "mail.test.activity,false,list":
+                    '<tree string="MailTestActivity"><field name="name"/></tree>',
+            });
+
+            const { env, openView } = await start({
+                serverData,
+            });
+            await openView({
+                res_model: "mail.test.activity",
+                views: [[false, "activity"]],
+            });
+            patchWithCleanup(env.services.action, {
+                doAction(action, options) {
+                    assert.step("doAction");
+                    options.onClose();
+                },
+            });
+
+            // open search create dialog and schedule an activity
+            await click(document.querySelector(".o_activity_view .o_record_selector"));
+            await click(document.querySelectorAll(".modal-lg .o_data_row .o_data_cell")[0]);
+
+            // again open search create dialog
+            await click(document.querySelector(".o_activity_view .o_record_selector"));
+            assert.verifySteps(["[]", "doAction", "[]"]);
+        }
+    );
+
+    QUnit.test(
+        "activity view: 'onClose' of 'openActivityFormView' does not add activity_ids condition as selectCreateDialog domain",
+        async function (assert) {
+            patchWithCleanup(ActivityController.prototype, {
+                openActivityFormView(resId, activityTypeId) {
+                    this._super(resId, activityTypeId);
+                    assert.step(JSON.stringify(this.props.domain));
+                },
+            });
+            Object.assign(serverData.views, {
+                "mail.test.activity,false,list":
+                    '<tree string="MailTestActivity"><field name="name"/></tree>',
+            });
+
+            const { env, openView } = await start({
+                serverData,
+            });
+            await openView({
+                res_model: "mail.test.activity",
+                views: [[false, "activity"]],
+            });
+            patchWithCleanup(env.services.action, {
+                doAction(action, options) {
+                    assert.step("doAction");
+                    options.onClose();
+                },
+            });
```

Comment:
all of this is redundant for the 3 tests, can we make an helper that create the view and patch it ?